src/app/SubscriptionResumptionStorage.h

```diff
@@ -0,0 +1,170 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the interface to store subscription information.
+ */
+
+#pragma once
+
+#include <app/ReadClient.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/support/CommonIterator.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * The SubscriptionResumptionStorage interface is used to persist subscriptions when they are established.
+ */
+class SubscriptionResumptionStorage
+{
+public:
+    // Structs to hold
+    struct AttributePathParamsValues
+    {
+        ClusterId mClusterId;
+        AttributeId mAttributeId;
+        EndpointId mEndpointId;
+        void SetValues(AttributePathParams & params)
+        {
+            mEndpointId  = params.mEndpointId;
+            mClusterId   = params.mClusterId;
+            mAttributeId = params.mAttributeId;
+        }
+        AttributePathParams GetParams() { return AttributePathParams(mEndpointId, mClusterId, mAttributeId); }
+    };
+    struct EventPathParamsValues
+    {
+        ClusterId mClusterId;
+        EventId mEventId;
+        EndpointId mEndpointId;
+        bool mIsUrgentEvent;
+        void SetValues(EventPathParams & params)
+        {
+            mEndpointId    = params.mEndpointId;
+            mClusterId     = params.mClusterId;
+            mEventId       = params.mEventId;
+            mIsUrgentEvent = params.mIsUrgentEvent;
+        }
+        EventPathParams GetParams() { return EventPathParams(mEndpointId, mClusterId, mEventId, mIsUrgentEvent); }
+    };
+
+    /**
+     * Struct to hold information about subscriptions
+     */
+    struct SubscriptionInfo
+    {
+        NodeId mNodeId;
+        FabricIndex mFabricIndex;
+        SubscriptionId mSubscriptionId;
+        uint16_t mMinInterval;
```

Comment:
Put unit of measurement her. Probably `mMinIntervalSec` or similar. Same for max interval.

General coding style I have seen for structures is to not use `m` prefix and use that for classes. not sure how much we care here.

Reply:
I've seen structs with `m` prefix and without. I'm okay with either one.

As for the interval members - the names are directly copied from ReadHandler. I can make the change but in that case, should I change all instances of them in ReadHandler as well? @andy31415